vertx-web/src/test/java/io/vertx/ext/web/RouterTest.java

```diff
@@ -1855,4 +1856,35 @@ public void testGetWithPlusPath2() throws Exception {
     });
     testRequest(HttpMethod.GET, "/some+path?q1=some+query", 200, "foo");
   }
+
+  @Test
+  public void testMethodNotAllowed() throws Exception {
+    router.get("/abc").handler(context -> {
+      fail("Should fail!");
+    });
+
+    testRequest(HttpMethod.POST, "/abc", null, resp -> {
+      assertEquals(405, resp.statusCode());
+      MultiMap headers = resp.headers();
+      assertTrue(headers.contains("Allow"));
+      assertEquals("GET", headers.get("Allow"));
+    }, 405, "Method Not Allowed", null);
+  }
+
+  @Test
+  public void testMethodNotAllowed2() throws Exception {
+    router.clear();
+    router.route().path("/abc").method(HttpMethod.GET).method(HttpMethod.PUT).handler(context -> {
+      fail("Should fail!");
+    });
```

Comment:
Also test for:

	router.get("/abc").handler(context -> {
		fail("Should fail!");
	});
	router.put("/abc").handler(context -> {
		fail("Should fail!");
	});

If I do a `PUT` request, it should work. If I do a `POST` request, it should fail with the info that only both GET and PUT are allowed (I'll ask for a couple more tests later when I'm back).